src/async_impl/response.rs

```diff
@@ -45,17 +51,29 @@ impl Response {
         &mut self.headers
     }
 
-    /// Get a mutable reference to the `Body` of this `Response`.
+    /// Get a readable response body.
+    /// 
+    /// The response will be decoded.
     #[inline]
-    pub fn body_mut(&mut self) -> &mut Body {
+    pub fn body_mut(&mut self) -> &mut Decoder {
         &mut self.body
     }
 
+    /// Get a readable response body.
+    /// 
+    /// This function will replace the body on the response with an empty one.
+    #[inline]
+    pub fn body(&mut self) -> Decoder {
```

Comment:
I think it's a bit unintuitive to have to use `mem::replace`, but I guess at some point if this uses `http::Response` then we'll have `into_parts`.